### PR TITLE
PAAS-2309: Update HAProxy config to not index technical domains

### PR DESCRIPTION
**Pull request by amitc-jahia**
JIRA issue: https://jira.jahia.org/browse/PAAS-2309

Short description:

### CODE REVIEW

File: assets/haproxy/robots_noindex.http
Comment (laurentfufu) [inline assets/haproxy/robots_noindex.http:1]: In fact in the ticket PAAS-2309 example I use the `.http` extension but that's something I should have removed. Indeed, the `.http` extension implies that the file also contains the HTTP code and the response headers which is not the case here.

File: packages/jahia/migrations/migrate-to-v23.yaml [hunk 1]
```diff
@@ -133,3 +134,16 @@ actions:
   removeAzurePythonPackages:
     - cmd[cp, proc]: pip list | awk '$1~/^(adal|azure)/ {print $1}' | xargs pip uninstall -y || true
       user: root
+
+  updateHaproxyConf:
+    - cmd[bl]: |-
+        HAPROXY_DIR=/etc/haproxy
+        HAPROXY_CONF_DIR=$HAPROXY_DIR/haproxy.cfg.jahia
+        HAPROXY_JAHIA_CONF=$HAPROXY_CONF_DIR/jahia-cloud.cfg
+        if ( ! grep -q "$HAPROXY_DIR/robots_noindex.http" $HAPROXY_JAHIA_CONF ); then
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v23.yaml:143]: cf my comment about the file name

**Review by laurentfufu — CHANGES_REQUESTED** [review]

File: assets/haproxy/robots_noindex.http [hunk 2]
```diff
@@ -0,0 +1 @@
+# ¯_(ツ)_/¯
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply assets/haproxy/robots_noindex.http]: Fixed

File: assets/haproxy/haproxy-jahia-cloud.cfg [hunk 3]
```diff
@@ -73,6 +73,12 @@ frontend ft_http
     #acl haproxy_adm_panel_path_beg  path_beg -i /haproxy_adm_panel #HTTP_AUTH_BASIC
     #http-request auth realm env-access-restricted if !auth_ok !tools !haproxy_adm_panel_path_beg #HTTP_AUTH_BASIC
 
+    acl is_robots_txt path_end /robots.txt
+    acl is_host_technical_domain hdr_reg(Host) ^.*\.(j|cloud)\.jahia.com$
+    acl is_x-forwarded-host_set hdr(X-Forwarded-Host) -m found
+    acl is_x-forwarded-host_technical_domain hdr_reg(X-Forwarded-Host) ^.*\.(j|cloud)\.jahia.com$ if is_x-forwarded-host_set
+    http-request return status 200 content-type "text/plain" file "/etc/haproxy/robots_noindex.http" hdr "cache-control" "no-cache" if is_robots_txt  is_host_technical_domain !is_x-forwarded-host_set || is_robots_txt is_host_technical_domain is_x-forwarded-host_technical_domain
```
[the remark this answers is not in the document]
Comment (amitc-jahia) [reply assets/haproxy/haproxy-jahia-cloud.cfg]: Fixed

File: assets/haproxy/robots_noindex.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# ¯_(ツ)_/¯
```
Comment (laurentfufu) [inline assets/haproxy/robots_noindex.txt:1]:
```suggestion
# ¯\_(ツ)_/¯
```
or
```suggestion
```
&nbsp;  
`¯\_(ツ)_/¯`

**Review by laurentfufu — CHANGES_REQUESTED** [review]

File: assets/haproxy/robots_noindex.txt (hunk 4 above)
Comment (amitc-jahia) [reply assets/haproxy/robots_noindex.txt]: Fixed

**Review by laurentfufu — DISMISSED** [review]

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**